web/src/hooks/useUserLocation.ts

```diff
@@ -43,18 +43,21 @@ export const getUserLocation = async (): Promise<UserLocationType> =>
     )
   })
 
-const useUserLocation = (): { data: UserLocationType | null; refresh: () => void } => {
-  const [refreshState, setRefreshState] = useState(false)
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const { data } = useLoadAsync(useCallback(async () => getUserLocation(), [refreshState]))
+type useUserLocationReturn = {
+  data: UserLocationType | null
+  refresh: () => void
+  error: Error | null
+  loading: boolean
+}
 
-  const refresh = useCallback(() => {
-    setRefreshState(prev => !prev)
-  }, [])
+const useUserLocation = (): useUserLocationReturn => {
+  const { data, refresh, error, loading } = useLoadAsync(useCallback(async () => getUserLocation(), []))
```

Comment:
:upside_down_face: Why not just
```suggestion
  return useLoadAsync(useCallback(getUserLocation, []))
```

Reply:
The callback isn't even needed as `getUserLocation` is not defined inside the hook. It would only be needed if you did the following as I thought would make sense (but perhaps this is better):
``` ts
return useLoadAsync(
    useCallback(async () => {
      const userLocation = await getUserLocation()
      return userLocation.status === 'ready' ? userLocation.coordinates : null
    }, []),
  )
```